examples/powercli/hostmaint-alarms/handler/script.ps1

```diff
@@ -12,25 +12,30 @@ if($env:function_debug -eq "true") {
 }
 
 $eventObjectName = $json.data.host.name
-$managedObjectReference = $json.data.host.type
 
 Set-PowerCLIConfiguration -InvalidCertificateAction Ignore  -DisplayDeprecationWarnings $false -ParticipateInCeip $false -Confirm:$false | Out-Null
 
 # Connect to vCenter Server
 Write-Host "Connecting to vCenter Server ..."
 Connect-VIServer -Server $($VC_CONFIG.VC) -User $($VC_CONFIG.VC_USERNAME) -Password $($VC_CONFIG.VC_PASSWORD)
 
+# Construct MoRef from Type/Value
+$moRef = New-Object VMware.Vim.ManagedObjectReference
+$moRef.Type = $json.data.host.host.type
+$moRef.Value = $json.data.host.host.Value
+$hostMoRef = Get-View $moRef
+
 # Get the vCenter AlarmManager
 $alarmManager = Get-View AlarmManager
-if ($json.topic -eq "entered.maintenance.mode") {
+if ($json.subject -eq "EnteredMaintenanceModeEvent") {
     # Disable alarm actions on the host
     Write-Host "Disabling alarm actions on host: $eventObjectName"
-    $alarmManager.EnableAlarmActions($managedObjectReference, $false)
+    $alarmManager.EnableAlarmActions($hostMoRef.MoRef, $false)
 }
 else {
```

Comment:
Suggestion: in order to simplify this example how about just deploying on function (it's the same function anyways), subscribing to these two topics (Enter/Exit) and handling that in the logic?

Either way, to further harden the logic I'd add an `-eq` check for the `else` case as well checking for `ExitMaintenanceModeEvent`. If someone messes up the event subscription, this could cause alarms being enabled for any other event (as per the trigger annotation).

Reply:
Good suggestion! Consolidated both